agent/engine/docker_task_engine.go

```diff
@@ -489,28 +489,38 @@ func (engine *DockerTaskEngine) handleDockerEvents(ctx context.Context) {
 // handleDockerEvent is responsible for taking an event that correlates to a
 // container and placing it in the context of the task to which that container
 // belongs.
-func (engine *DockerTaskEngine) handleDockerEvent(event DockerContainerChangeEvent) bool {
-	log.Debug("Handling a docker event", "event", event)
+func (engine *DockerTaskEngine) handleDockerEvent(event DockerContainerChangeEvent) {
+	seelog.Debugf("Handling a docker event, event: %s", event)
 
 	task, taskFound := engine.state.TaskByID(event.DockerID)
 	cont, containerFound := engine.state.ContainerByID(event.DockerID)
 	if !taskFound || !containerFound {
-		log.Debug("Event for container not managed", "dockerId", event.DockerID)
-		return false
+		seelog.Debugf("Event for container not managed, container: %s", event.DockerID)
+		return
 	}
+
+	// Container health status change doesnot affect the container status
+	// no need to process this in task manager
+	if event.Type == api.ContainerHealthEvent {
+		if cont.Container.HealthCheckShouldBeReported() {
+			cont.Container.SetHealthStatus(event.DockerContainerMetadata.Health)
+		}
+		return
+	}
+
 	engine.processTasks.RLock()
 	managedTask, ok := engine.managedTasks[task.Arn]
 	// hold the lock until the message is sent so we don't send on a closed channel
 	defer engine.processTasks.RUnlock()
 	if !ok {
-		log.Crit("Could not find managed task corresponding to a docker event", "event", event, "task", task)
-		return true
+		seelog.Criticalf("Could not find managed task corresponding to a docker event, event: %s, task: %s", event, task)
```

Comment:
I don't think we need the whole task here (or in other log lines you've changed). Can you use `task.Arn` instead?